src/pylife/stress/rainflow/fkm_nonlinear.py

```diff
@@ -514,7 +528,8 @@ def _perform_hcm_algorithm(self, *, samples, recording_lists, largest_point, pre
         self._hcm_message += f"turning points: {samples}\n"
 
         # iterate over loads from the given list of samples
-        for current_load in load_turning_points:
+        li = load_turning_points.index.to_frame()['load_step']
+        for _, current_load in load_turning_points.groupby((li!=li.shift()).cumsum(), sort=False):
```

Comment:
The `li!=li.shift()` and `load_turning_points.groupby((li!=li.shift()).cumsum()` could be separate statements with variable assignment to make it better readable